builder/builder.go

```diff
@@ -348,9 +352,18 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 	b.builtMain = true
 
 	if opt.Push && opt.OnlyArtifact == nil && !opt.OnlyFinalTargetImages {
-		err = b.s.buildMainMulti(ctx, bf, onImage, onArtifact, onFinalArtifact, "--push")
-		if err != nil {
-			return nil, errors.Wrapf(err, "build push")
+		hasRunPush := false
+		for _, sts := range mts.All() {
+			if sts.RunPush.Initialized {
+				hasRunPush = true
+				break
+			}
+		}
+		if hasRunPush {
+			err = b.s.buildMainMulti(ctx, bf, onImage, onArtifact, onFinalArtifact, "--push")
+			if err != nil {
+				return nil, errors.Wrapf(err, "build push")
+			}
```

Comment:
This should be a no-op (I think) - we're avoiding executing run-pushes if no run push was defined anywhere.